modules/core/04-channel/types/upgrade.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+)
+
+// ValidateBasic performs a basic validation of the upgrade fields
+func (u Upgrade) ValidateBasic() error {
+	if err := u.ProposedUpgrade.ValidateBasic(); err != nil {
+		return errorsmod.Wrap(err, "proposed upgrade fields are invalid")
+	}
+
+	if !u.Timeout.IsValid() {
+		return errorsmod.Wrap(ErrInvalidUpgrade, "upgrade timeout cannot be empty")
+	}
+
+	// TODO: determine if last packet sequence sent can be 0?
+	return nil
+}
+
+// ValidateBasic performs a basic validation of the proposed upgrade fields
+func (muf ModifiableUpgradeFields) ValidateBasic() error {
+	if !(muf.Ordering == ORDERED || muf.Ordering == UNORDERED) {
```

Comment:
can we use the contains function here too?

`collections.Contains(muf.Ordering, []Order{ORDERED, UNORDERED})`